packages/server-wallet/src/utilities/__test__/signatures.test.ts

```diff
@@ -32,3 +32,26 @@ it('sign vs fastSign', async () => {
     }
   });
 });
+
+it('getSignerAddress vs fastRecover', async () => {
+  _.range(5).map(async channelNonce => {
+    const {address, privateKey} = Wallet.createRandom();
+    const state: State = {
+      chainId: '0x1',
+      channelNonce,
+      participants: [participant({signingAddress: address})],
+      outcome: simpleEthAllocation([]),
+      turnNum: 1,
+      isFinal: false,
+      appData: '0x0',
+      appDefinition: AddressZero,
+      challengeDuration: 0x5,
+    };
+
+    const signedState = await fastSignState(state, privateKey);
+
+    const recovered = getSignerAddress(signedState.state, signedState.signature);
+    const fastRecovered = fastRecoverAddress(signedState.state, signedState.signature);
+    expect(recovered).toEqual(fastRecovered);
```

Comment:
The one issue with this is if we ever do find a test failure, we'll have the recovered address but not the private key, so it would be hard to reproduce the test failure :P

In the corresponding test for `fastSignState`, we log the private key in case of an error (which should theoretically never happen)

Reply:
Updated 7b56ee22d5d11d3963e6e54a9f27428cc083bf9c